ledger/test-common/src/main/scala/com/digitalasset/platform/testing/TimeBoundObserver.scala

```diff
@@ -10,19 +10,24 @@ import io.grpc.stub.StreamObserver
 import scala.concurrent.duration.FiniteDuration
 import scala.concurrent.{ExecutionContext, Future, Promise}
 
-final class TimeBoundObserver[T](duration: FiniteDuration)(
+final class TimeBoundObserver[T](duration: FiniteDuration, maximumResults: Option[Int] = None)(
     implicit executionContext: ExecutionContext)
     extends StreamObserver[T] {
 
   private val promise = Promise[Vector[T]]
   private val buffer = Vector.newBuilder[T]
+  private var size = 0
 
   Delayed.by(duration)(onCompleted())
 
   def result: Future[Vector[T]] = promise.future
 
   override def onNext(value: T): Unit = {
     buffer += value
+    size += 1
```

Comment:
`buffer` is a vector builder, it should be fine to just use `buffer.size` instead of keep track of it. It doesn't look like this can cause any problem, but I can see how keeping track of the thing separately can cause weird concurrency bugs in future revisions.

Reply:
Unfortunately, `buffer.size` is not a thing. The same logic is in `SizeBoundObserver`.